### PR TITLE
Windows: impl `create symlink`

### PR DESCRIPTION
**Pull request by JunkuiZhang**
The `symlink` on windows is quiet annoying.

Release Notes:

- N/A

### CODE REVIEW

File: crates/fs/src/fs.rs [hunk 1]
```diff
@@ -1519,6 +1519,100 @@ async fn file_id(path: impl AsRef<Path>) -> Result<u64> {
     .await
 }
 
+#[cfg(target_os = "windows")]
+async fn windows_create_symlink(target: PathBuf, path: &Path) -> Result<()> {
```
Comment (zaucy) [inline crates/fs/src/fs.rs:1523]: does using `std::os::windows::fs::{symlink_file, symlink_dir}` not work for some reason?

Comment (kazatsuyu) [reply crates/fs/src/fs.rs:1523]: AFAIK, It does not work without developer mode

Comment (JunkuiZhang) [reply crates/fs/src/fs.rs:1532]: Yes, creating symbolic links on Windows requires the corresponding permissions, either by enabling Developer Mode on Windows or by launching Zed as `Administrator`

**Comment by mikayla-maki** [comment]
Does this implementation mean that you'll have to be re-prompted for each symlink invocation?

**Comment by zaucy** [comment]
> > Does this implementation mean that you'll have to be re-prompted for each symlink invocation?
> 
> Yes.

That's a real bummer. I think it would be better if Zed tried to make a symlink and they don't have developer mode on we prompt them to enable developer mode.

Thoughts?

**Comment by kazatsuyu** [comment]
AFAIK, the use case of it is just only [here](https://github.com/zed-industries/zed/blob/80bc6c8cc800e23ba79723b0c46b731a62203224/crates/extension/src/extension_store.rs#L571). I think there is no matter to request developer mode to extension developers.

However, it seems to enough to use junction instead of symlink.

**Comment by JunkuiZhang** [comment]
Now, the process initiated when creating symbolic links will remain running in the background and exit when Zed is closed. To demonstrate this, I added some code in Zed to continuously create 10 symbolic links when Zed is launched. Please note that the UAC pop-up window in Windows cannot be recorded in the video.


https://github.com/zed-industries/zed/assets/14981363/9dad708e-721b-477c-9f1f-47e95e5dea73

**Comment by mikayla-maki** [comment]
Thank you for exploring this implementation @JunkuiZhang, I think we'd prefer to just require that developer mode be enabled for anyone using extensions. Maybe open a new PR using the `SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE` flag, without the spare process. Maybe in a follow up PR we can adjust how extensions are initialized if a user runs Zed in non-symlink mode.

**Pull request closed without merging**